Installation issue: gtkplus fails to build

### Steps to reproduce the issue

tried building versions with autotool and meson, both failed.
```console
$ spack spec gtkplus@3.24.29 arch=x86_64 %gcc@11.5.0
 -   gtkplus@3.24.29~cups build_system=autotools arch=linux-rhel9-x86_64
[+]      ^at-spi2-core@2.48.3~strip build_system=meson buildtype=release default_library:=shared arch=linux-rhel9-x86_64
[+]          ^compiler-wrapper@1.0 build_system=generic arch=linux-rhel9-x86_64
[+]          ^dbus@1.16.2~strip~xml_docs build_system=meson buildtype=release default_library:=shared system-socket=default arch=linux-rhel9-x86_64
[+]              ^docbook-xml@4.5 build_system=generic arch=linux-rhel9-haswell
[+]                  ^libxml2@2.13.5+http+pic~python+shared build_system=autotools arch=linux-rhel9-x86_64
[+]              ^docbook-xsl@1.79.2 build_system=generic patches:=a92c397 arch=linux-rhel9-haswell
[+]              ^expat@2.7.1+libbsd build_system=autotools arch=linux-rhel9-x86_64
[+]                  ^libbsd@0.12.2 build_system=autotools arch=linux-rhel9-x86_64
[+]                      ^libmd@1.1.0 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^libsm@1.2.6 build_system=autotools arch=linux-rhel9-x86_64
[+]                  ^libice@1.1.2 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^libx11@1.8.11 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^inputproto@2.3.2 build_system=autotools arch=linux-rhel9-haswell
[+]              ^kbproto@1.0.7 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^perl@5.40.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-rhel9-x86_64
[+]                  ^zlib-ng@2.2.4+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-rhel9-x86_64
[+]              ^xtrans@1.6.0 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^libxtst@1.2.5 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^meson@1.7.0 build_system=python_pip patches:=0f0b1bd arch=linux-rhel9-x86_64
[+]              ^py-pip@25.0.1 build_system=generic arch=linux-rhel9-x86_64
[+]              ^py-setuptools@62.6.0 build_system=generic arch=linux-rhel9-x86_64
[+]              ^py-wheel@0.37.1 build_system=generic arch=linux-rhel9-x86_64
[+]          ^ninja@1.12.1+re2c build_system=generic patches:=93f4bb3 arch=linux-rhel9-x86_64
[+]              ^python@3.11.11+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches:=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-rhel9-x86_64
[+]                  ^gettext@0.23.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-rhel9-x86_64
[+]                      ^tar@1.35 build_system=autotools zip=pigz arch=linux-rhel9-x86_64
[+]                          ^pigz@2.8 build_system=makefile arch=linux-rhel9-x86_64
[+]                  ^openssl@3.4.1~docs+shared build_system=generic certs=mozilla arch=linux-rhel9-x86_64
[+]                  ^sqlite@3.46.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-rhel9-x86_64
[+]              ^re2c@3.1 build_system=autotools arch=linux-rhel9-haswell
[+]          ^python@3.11.11+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix+tkinter+uuid+zlib build_system=generic patches:=b0615b2,ebdca64,f2fd060 arch=linux-rhel9-x86_64
[+]              ^gdbm@1.23 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^libxcrypt@4.4.38~obsolete_api build_system=autotools arch=linux-rhel9-x86_64
[+]              ^openssl@3.4.1~docs+shared build_system=generic certs=mozilla arch=linux-rhel9-x86_64
[+]                  ^ca-certificates-mozilla@2025-02-25 build_system=generic arch=linux-rhel9-haswell
[+]              ^readline@8.2 build_system=autotools patches:=1ea4349,24f587b,3d9885e,5911a5b,622ba38,6c8adf8,758e2ec,79572ee,a177edc,bbf97f1,c7b45ff,e0013d9,e065038 arch=linux-rhel9-x86_64
[+]              ^sqlite@3.46.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-rhel9-x86_64
[+]              ^tcl@8.6.12 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^tk@8.6.11+xft+xss build_system=autotools arch=linux-rhel9-x86_64
[+]                  ^libxscrnsaver@1.2.4 build_system=autotools arch=linux-rhel9-x86_64
[+]                      ^scrnsaverproto@1.2.2 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^util-linux-uuid@2.41 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^recordproto@1.14.2 build_system=autotools arch=linux-rhel9-x86_64
[+]      ^atk@2.38.0~strip build_system=meson buildtype=release default_library:=shared arch=linux-rhel9-x86_64
[+]          ^libffi@3.4.7 build_system=autotools arch=linux-rhel9-x86_64
[+]      ^cairo@1.17.4+X+fc+ft+gobject+pdf+pic+png+shared~svg build_system=autotools patches:=7097196 arch=linux-rhel9-x86_64
[+]          ^autoconf@2.72 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^automake@1.16.5 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^fontconfig@2.13.1~pic build_system=autotools arch=linux-rhel9-x86_64
[+]              ^font-util@1.4.1 build_system=autotools fonts:=encodings,font-adobe-100dpi,font-adobe-75dpi,font-adobe-utopia-100dpi,font-adobe-utopia-75dpi,font-adobe-utopia-type1,font-alias,font-arabic-misc,font-bh-100dpi,font-bh-75dpi,font-bh-lucidatypewriter-100dpi,font-bh-lucidatypewriter-75dpi,font-bh-type1,font-bitstream-100dpi,font-bitstream-75dpi,font-bitstream-speedo,font-bitstream-type1,font-cronyx-cyrillic,font-cursor-misc,font-daewoo-misc,font-dec-misc,font-ibm-type1,font-isas-misc,font-jis-misc,font-micro-misc,font-misc-cyrillic,font-misc-ethiopic,font-misc-meltho,font-misc-misc,font-mutt-misc,font-schumacher-misc,font-screen-cyrillic,font-sun-misc,font-winitzki-cyrillic,font-xfree86-type1 arch=linux-rhel9-x86_64
[+]                  ^autoconf@2.72 build_system=autotools arch=linux-rhel9-x86_64
[+]                  ^automake@1.16.5 build_system=autotools arch=linux-rhel9-x86_64
[+]                  ^bdftopcf@1.1.1 build_system=autotools arch=linux-rhel9-x86_64
[+]                      ^fontsproto@2.1.3 build_system=autotools arch=linux-rhel9-x86_64
[+]                      ^libxfont@1.5.4 build_system=autotools arch=linux-rhel9-x86_64
[+]                  ^mkfontdir@1.0.7 build_system=autotools arch=linux-rhel9-x86_64
[+]                  ^mkfontscale@1.2.3 build_system=autotools arch=linux-rhel9-x86_64
[+]                      ^freetype@2.13.2+pic+shared build_system=autotools arch=linux-rhel9-x86_64
[+]                          ^libpng@1.6.47~ipo~pic build_system=cmake build_type=Release generator=make libs:=shared,static arch=linux-rhel9-x86_64
[+]                      ^libfontenc@1.1.8 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^gperf@3.1 build_system=autotools patches:=3dd36db arch=linux-rhel9-haswell
[+]          ^freetype@2.13.2+pic+shared build_system=autotools arch=linux-rhel9-x86_64
[+]          ^libpng@1.6.47~ipo~pic build_system=cmake build_type=Release generator=make libs:=shared,static arch=linux-rhel9-x86_64
[+]              ^cmake@3.31.6~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release arch=linux-rhel9-x86_64
[+]          ^libtool@2.4.7 build_system=autotools arch=linux-rhel9-haswell
[+]              ^findutils@4.10.0 build_system=autotools patches:=440b954 arch=linux-rhel9-haswell
[+]          ^libxcb@1.17.0 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^libpthread-stubs@0.5 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^libxau@1.0.12 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^libxdmcp@1.1.5 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^xcb-proto@1.17.0 build_system=autotools arch=linux-rhel9-x86_64
[+]                  ^python-venv@1.0 build_system=generic arch=linux-rhel9-x86_64
[+]          ^libxext@1.3.6 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^libxrender@0.9.12 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^renderproto@0.11.1 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^m4@1.4.19+sigsegv build_system=autotools patches:=9dc5fbd,bfdffa7 arch=linux-rhel9-x86_64
[+]              ^diffutils@3.10 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^libsigsegv@2.14 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^pixman@0.44.0+shared~strip build_system=meson buildtype=release default_library:=shared arch=linux-rhel9-x86_64
[+]          ^which@2.21 build_system=autotools arch=linux-rhel9-haswell
[+]      ^compiler-wrapper@1.0 build_system=generic arch=linux-rhel9-haswell
[+]      ^fixesproto@5.0 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^util-macros@1.20.1 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^xextproto@7.3.0 build_system=autotools arch=linux-rhel9-x86_64
[+]      ^fribidi@1.0.12 build_system=autotools arch=linux-rhel9-x86_64
[e]      ^gcc@11.5.0~binutils+bootstrap~graphite~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' arch=linux-rhel9-haswell
[+]      ^gcc-runtime@11.5.0 build_system=generic arch=linux-rhel9-x86_64
[+]      ^gdk-pixbuf@2.42.12~man~strip~tiff build_system=meson buildtype=release default_library:=shared arch=linux-rhel9-x86_64
[+]          ^libjpeg-turbo@3.0.4~ipo~jpeg8~partial_decoder+pic build_system=cmake build_type=Release generator=make libs:=shared,static arch=linux-rhel9-x86_64
[+]              ^cmake@3.31.6~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release arch=linux-rhel9-x86_64
[+]                  ^curl@8.11.1~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=openssl arch=linux-rhel9-x86_64
[+]              ^nasm@2.16.03 build_system=autotools arch=linux-rhel9-haswell
[+]          ^zlib@1.3.1+optimize+pic+shared build_system=makefile arch=linux-rhel9-x86_64
[+]      ^gettext@0.23.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-rhel9-x86_64
[+]          ^bzip2@1.0.8~debug~pic+shared build_system=generic arch=linux-rhel9-x86_64
[+]          ^libiconv@1.18 build_system=autotools libs:=shared,static arch=linux-rhel9-x86_64
[+]          ^libxml2@2.13.5+http+pic~python+shared build_system=autotools arch=linux-rhel9-x86_64
[+]          ^ncurses@6.5~symlinks+termlib abi=none build_system=autotools patches:=7a351bc arch=linux-rhel9-x86_64
[+]          ^tar@1.35 build_system=autotools zip=pigz arch=linux-rhel9-x86_64
[+]              ^pigz@2.8 build_system=makefile arch=linux-rhel9-x86_64
[+]              ^zstd@1.5.7+programs build_system=makefile compression:=none libs:=shared,static arch=linux-rhel9-x86_64
[+]          ^xz@5.6.3~pic build_system=autotools libs:=shared,static arch=linux-rhel9-x86_64
[+]      ^glib@2.82.5~libmount~strip build_system=meson buildtype=release default_library:=shared tracing:=none arch=linux-rhel9-x86_64
[+]          ^elfutils@0.192~debuginfod+exeprefix+nls build_system=autotools arch=linux-rhel9-x86_64
[+]          ^pcre2@10.44~jit+multibyte+pic build_system=autotools arch=linux-rhel9-x86_64
[+]          ^perl@5.40.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-rhel9-x86_64
[+]              ^berkeley-db@18.1.40+cxx~docs+stl build_system=autotools patches:=26090f4,b231fcc arch=linux-rhel9-x86_64
[+]          ^python-venv@1.0 build_system=generic arch=linux-rhel9-x86_64
[e]      ^glibc@2.34 build_system=autotools arch=linux-rhel9-x86_64
[+]      ^gmake@4.4.1~guile build_system=generic arch=linux-rhel9-x86_64
[+]      ^gobject-introspection@1.78.1~strip build_system=meson buildtype=release default_library:=shared arch=linux-rhel9-x86_64
[+]          ^bison@3.8.2~color build_system=autotools arch=linux-rhel9-x86_64
[+]          ^flex@2.6.3+lex~nls build_system=autotools arch=linux-rhel9-haswell
[+]      ^inputproto@2.3.2 build_system=autotools arch=linux-rhel9-x86_64
[+]      ^libepoxy@1.5.10+glx~strip build_system=meson buildtype=release default_library:=shared arch=linux-rhel9-x86_64
[+]          ^glx@1.4 build_system=bundle arch=linux-rhel9-x86_64
[+]              ^mesa@23.3.6+glx~llvm+opengl~opengles+osmesa~strip build_system=meson buildtype=release default_library:=shared arch=linux-rhel9-x86_64
[+]                  ^glproto@1.4.17 build_system=autotools arch=linux-rhel9-x86_64
[+]                  ^libunwind@1.8.1~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs~pic+tests+weak_backtrace~xz~zlib build_system=autotools components:=none libs:=shared,static arch=linux-rhel9-x86_64
[+]                  ^libxt@1.3.1 build_system=autotools arch=linux-rhel9-x86_64
[+]                  ^py-mako@1.2.4 build_system=python_pip arch=linux-rhel9-x86_64
[+]                      ^py-markupsafe@2.1.3 build_system=python_pip arch=linux-rhel9-x86_64
[+]      ^librsvg@2.58.2~doc build_system=autotools arch=linux-rhel9-x86_64
[+]          ^harfbuzz@10.2.0+graphite2~strip build_system=meson buildtype=release default_library:=shared arch=linux-rhel9-x86_64
[+]              ^graphite2@1.3.14~ipo build_system=cmake build_type=Release generator=make patches:=39e58d6 arch=linux-rhel9-x86_64
[+]              ^icu4c@76.1 build_system=autotools cxxstd=17 arch=linux-rhel9-x86_64
[+]          ^py-docutils@0.20.1 build_system=python_pip arch=linux-rhel9-x86_64
[+]              ^py-setuptools@68.2.2 build_system=generic arch=linux-rhel9-x86_64
[+]          ^rust@1.85.0+dev~docs+src build_system=generic arch=linux-rhel9-x86_64
[+]              ^curl@8.11.1~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=openssl arch=linux-rhel9-x86_64
[+]                  ^nghttp2@1.65.0 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^libgit2@1.8.0~curl~ipo+mmap+ssh build_system=cmake build_type=Release generator=make https=system arch=linux-rhel9-x86_64
[+]                  ^pcre@8.45~jit+multibyte+pic+shared+static+utf build_system=autotools arch=linux-rhel9-x86_64
[+]              ^libssh2@1.11.1+shared build_system=autotools crypto=openssl arch=linux-rhel9-x86_64
[+]              ^rust-bootstrap@1.85.0 build_system=generic arch=linux-rhel9-x86_64
[+]                  ^patchelf@0.17.2 build_system=autotools arch=linux-rhel9-x86_64
[+]      ^libxi@1.7.10 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^libxfixes@5.0.3 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^xproto@7.0.31 build_system=autotools arch=linux-rhel9-x86_64
[+]      ^libxkbcommon@1.7.0~strip~wayland build_system=meson buildtype=release default_library:=shared arch=linux-rhel9-x86_64
[+]          ^xkeyboard-config@2.34 build_system=autotools arch=linux-rhel9-x86_64
[+]              ^intltool@0.51.0 build_system=autotools patches:=ca9d656 arch=linux-rhel9-x86_64
[+]                  ^perl-xml-parser@2.47 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-libwww-perl@6.68 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-clone@0.46 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-encode-locale@1.05 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-file-listing@6.16 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-html-parser@3.72 build_system=perl arch=linux-rhel9-x86_64
[+]                              ^perl-html-tagset@3.24 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-http-cookies@6.11 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-http-daemon@6.16 build_system=perl arch=linux-rhel9-x86_64
[+]                              ^perl-module-build-tiny@0.048 build_system=perl arch=linux-rhel9-x86_64
[+]                                  ^perl-extutils-config@0.010 build_system=perl arch=linux-rhel9-x86_64
[+]                                  ^perl-extutils-helpers@0.026 build_system=perl arch=linux-rhel9-x86_64
[+]                                  ^perl-extutils-installpaths@0.013 build_system=perl arch=linux-rhel9-x86_64
[+]                                  ^perl-module-build@0.4234 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-http-date@6.06 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-http-message@6.45 build_system=perl arch=linux-rhel9-x86_64
[+]                              ^perl-io-html@1.004 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-http-negotiate@6.01 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-lwp-mediatypes@6.04 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-net-http@6.23 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-try-tiny@0.31 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-uri@5.08 build_system=perl arch=linux-rhel9-x86_64
[+]                              ^perl-test-needs@0.002010 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-www-robotrules@6.02 build_system=perl arch=linux-rhel9-x86_64
[+]              ^libxslt@1.1.42+crypto~python build_system=autotools arch=linux-rhel9-x86_64
[+]                  ^libgcrypt@1.11.0 build_system=autotools patches:=1ecb4ee,4e308ba arch=linux-rhel9-x86_64
[+]                      ^libgpg-error@1.51 build_system=autotools arch=linux-rhel9-x86_64
[+]                          ^gawk@5.3.1~nls build_system=autotools arch=linux-rhel9-x86_64
[+]                              ^gmp@6.3.0+cxx build_system=autotools libs:=shared,static arch=linux-rhel9-x86_64
[+]                              ^mpfr@4.2.1 build_system=autotools libs:=shared,static arch=linux-rhel9-x86_64
[+]                                  ^autoconf-archive@2023.02.20 build_system=autotools arch=linux-rhel9-haswell
[+]                                  ^texinfo@7.1 build_system=autotools arch=linux-rhel9-x86_64
[+]      ^pango@1.54.0+X~strip build_system=meson buildtype=release default_library:=shared arch=linux-rhel9-x86_64
[+]          ^json-glib@1.6.6~strip build_system=meson buildtype=release default_library:=shared arch=linux-rhel9-x86_64
[+]          ^libxft@2.3.8 build_system=autotools arch=linux-rhel9-x86_64
[+]      ^pkgconf@2.3.0 build_system=autotools arch=linux-rhel9-x86_64
[+]      ^shared-mime-info@1.10 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^intltool@0.51.0 build_system=autotools patches:=ca9d656 arch=linux-rhel9-x86_64
[+]              ^perl-xml-parser@2.47 build_system=perl arch=linux-rhel9-x86_64
[+]                  ^perl-libwww-perl@6.68 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-clone@0.46 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-encode-locale@1.05 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-file-listing@6.16 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-html-parser@3.72 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-html-tagset@3.24 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-http-cookies@6.11 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-http-daemon@6.16 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-module-build-tiny@0.048 build_system=perl arch=linux-rhel9-x86_64
[+]                              ^perl-extutils-config@0.010 build_system=perl arch=linux-rhel9-x86_64
[+]                              ^perl-extutils-helpers@0.026 build_system=perl arch=linux-rhel9-x86_64
[+]                              ^perl-extutils-installpaths@0.013 build_system=perl arch=linux-rhel9-x86_64
[+]                              ^perl-module-build@0.4234 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-http-date@6.06 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-http-message@6.45 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-io-html@1.004 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-http-negotiate@6.01 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-lwp-mediatypes@6.04 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-net-http@6.23 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-try-tiny@0.31 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-uri@5.08 build_system=perl arch=linux-rhel9-x86_64
[+]                          ^perl-test-needs@0.002010 build_system=perl arch=linux-rhel9-x86_64
[+]                      ^perl-www-robotrules@6.02 build_system=perl arch=linux-rhel9-x86_64
[+]      ^xrandr@1.5.3 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^libxrandr@1.5.4 build_system=autotools arch=linux-rhel9-x86_64
[+]          ^randrproto@1.5.0 build_system=autotools arch=linux-rhel9-x86_64

```


### Error message

<details><summary>Error message</summary>

<pre>
gtktestutils.c: In function 'gtk_test_register_all_types':
gtktestutils.c:767:10: fatal error: gtktypefuncs.inc: No such file or directory
  767 | #include "gtktypefuncs.inc"
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[3]: *** [Makefile:7303: libgtk_3_la-gtktestutils.lo] Error 1
make[3]: Leaving directory '/mnt/local/cache/sbulut/spack-stage/spack-stage-gtkplus-3.24.29-cjzeeqrjdtrdfmvsr3kgoa6gnnhw4akl/spack-src/gtk'
make[2]: *** [Makefile:3178: all] Error 2
make[2]: Leaving directory '/mnt/local/cache/sbulut/spack-stage/spack-stage-gtkplus-3.24.29-cjzeeqrjdtrdfmvsr3kgoa6gnnhw4akl/spack-src/gtk'
make[1]: *** [Makefile:751: all-recursive] Error 1
make[1]: Leaving directory '/mnt/local/cache/sbulut/spack-stage/spack-stage-gtkplus-3.24.29-cjzeeqrjdtrdfmvsr3kgoa6gnnhw4akl/spack-src'
make: *** [Makefile:624: all] Error 2

</pre></details>


### Information on your system

$ spack debug report
* **Spack:** 1.0.0.dev0 (99d06b95a3635055b90772de2cfb355bbf9f3bb6)
* **Python:** 3.9.21
* **Platform:** linux-rhel9-haswell


### Additional information

[spack-build-03-build-out.txt](https://github.com/user-attachments/files/20377315/spack-build-03-build-out.txt)

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

also, in the process I realized that cairo@1.16.0 isn't compiling:
```
  >> 88     perf/Makefile.sources:18: but option 'subdir-objects' is disabled
     89     perf/Makefile.am:3:   'perf/Makefile.sources' included from here
     90     test/Makefile.am:356: warning: user target 'recheck' defined here ...
     91     /disk/software/lib/hpcpm/spack_dev_2025_may_05_99d06b95a/opt/spack/linux-x86_64/automake-1.16.5-5khi7gky6ysxwtlhrtlkicndgfs5prrj/share/automake-1.16/am/check.am: ... overrides Automake target 'recheck' defined here
     92     autoreconf: Leaving directory '.'
     93     ==> cairo: Executing phase: 'configure'
     94     ==> [2025-05-22-10:26:18.503797] Find (max depth = None): ['/cache/sbulut/spack-stage/spack-stage-cairo-1.16.0-f7233fhfnmia7t7scpnziqxju7d5npbc/spack-src'] ['configure']

     ...

     396    checking whether cairo's SVG surface backend feature could be enabled... yes
     397    checking for LIBRSVG... no
     398    configure: WARNING: SVG backend will not be tested since librsvg >= 2.35.0 is not available
     399    checking for cairo's image surface backend feature...
     400    checking for pixman... no
     401    checking whether cairo's image surface backend feature could be enabled... no (requires pixman-1 >= 0.30.0 https://cairographics.org/releases/)
  >> 402    configure: error: mandatory image surface backend feature could not be enabled
```

also ghostscript fails to build:
```
==> Installing ghostscript-10.04.0-i3zwzefgcjstngiszc77hyvncoamqlv3 [109/110]
==> No binary for ghostscript-10.04.0-i3zwzefgcjstngiszc77hyvncoamqlv3 found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/c7/c764dfbb7b13fc71a7a05c634e014f9bb1fb83b899fe39efc0b6c3522a9998b1.tar.gz
    [100%]   89.19 MB @   32.0 MB/s
==> Ran patch() for ghostscript
==> ghostscript: Executing phase: 'autoreconf'
==> ghostscript: Executing phase: 'configure'
==> ghostscript: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    '/disk/software/lib/hpcpm/spack_dev_2025_may_05_99d06b95a/opt/spack/linux-x86_64/gmake-4.4.1-jo3moir43n3jfqtwez5dcrq72duxnzjz/bin/make' '-j16' 'V=1' 'default' 'so'

4 errors found in build log:
     3949    DEVICE_DEVS4= DEVICE_DEVS5= DEVICE_DEVS6= DEVICE_DEVS7= DEVICE_DEVS8= \
     3950    DEVICE_DEVS9= DEVICE_DEVS10= DEVICE_DEVS11= DEVICE_DEVS12= \
     3951    DEVICE_DEVS13= DEVICE_DEVS14= DEVICE_DEVS15= DEVICE_DEVS16= \
     3952    DEVICE_DEVS17= DEVICE_DEVS18= DEVICE_DEVS19= DEVICE_DEVS20= \
     3953    DEVICE_DEVS_EXTRA= \
     3954    /bin/sh <./soobj/ldt.tr
  >> 3955    /usr/bin/ld: ./obj/gdevopvp.o: undefined reference to symbol 'libiconv_open'
  >> 3956    /usr/bin/ld: /disk/software/lib/hpcpm/spack_dev_2025_may_05_99d06b95a/opt/spack/linux-x86_64/libiconv-1.18-6x7hjf4yc7bkihgnz6j5ie254crrf25j/lib/libiconv.so.2: error adding symbols: DSO missing from command line
  >> 3957    collect2: error: ld returned 1 exit status
  >> 3958    make: *** [/cache/sbulut/spack-stage/spack-stage-ghostscript-10.04.0-i3zwzefgcjstngiszc77hyvncoamqlv3/spack-src/base/unixlink.mak:172: bin/gs] Error 1
     3959    make: *** Waiting for unfinished jobs....
     3960    make[2]: Leaving directory '/mnt/local/cache/sbulut/spack-stage/spack-stage-ghostscript-10.04.0-i3zwzefgcjstngiszc77hyvncoamqlv3/spack-src'
     3961    /disk/software/lib/hpcpm/spack_dev_2025_may_05_99d06b95a/opt/spack/linux-x86_64/gmake-4.4.1-jo3moir43n3jfqtwez5dcrq72duxnzjz/bin/make -f Makefile DISPLAY_DEV=./soobj/display.dev BUILDDIRPREFIX=so GENOPT='' LDFLAGS=' '\
     3962     CFLAGS='-O2 -DNDEBUG -Wall -Wstrict-prototypes -Wundef -Wmissing-declarations -Wmissing-prototypes -Wwrite-strings -fno-strict-aliasing -Werror=declaration-after-statement -fno-builtin -fno-common -Werror=return-type -Wno-u
             nused-local-typedefs -DHAVE_STDINT_H=1 -DHAVE_DIRENT_H=1 -DHAVE_SYS_DIR_H=1 -DHAVE_SYS_TIME_H=1 -DHAVE_SYS_TIMES_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_LIBDL=1 -DGX_COLOR_INDEX_TYPE="unsigned long long" -D__USE_UNIX98=1 -DHAVE_SNPRI
             NTF  -DBUILD_PDF=1 -I/cache/sbulut/spack-stage/spack-stage-ghostscript-10.04.0-i3zwzefgcjstngiszc77hyvncoamqlv3/spack-src/pdf  -DHAVE_RESTRICT=1 -I/disk/software/lib/hpcpm/spack_dev_2025_may_05_99d06b95a/opt/spack/linux-x86_
             64/libtiff-4.7.0-fttn3acqpnl4tth5heaoeoyczojktgnp/include -I/disk/software/lib/hpcpm/spack_dev_2025_may_05_99d06b95a/opt/spack/linux-x86_64/zlib-1.3.1-ww5yrag6eb36voewk7js2fux4vjd2gfr/include -I/disk/software/lib/hpcpm/spack
             _dev_2025_may_05_99d06b95a/opt/spack/linux-x86_64/libjpeg-turbo-3.0.4-ljs6n22owttuzpoe4wsndqfwyrtfkb6w/include -I/disk/software/lib/hpcpm/spack_dev_2025_may_05_99d06b95a/opt/spack/linux-x86_64/zstd-1.5.7-prirs2fwtzrhjcktq5mo
             hxjjv7gz4l5s/include -fno-strict-aliasing -DHAVE_POPEN_PROTO=1  ' prefix=/disk/software/lib/hpcpm/spack_dev_2025_may_05_99d06b95a/opt/spack/linux-haswell/ghostscript-10.04.0-i3zwzefgcjstngiszc77hyvncoamqlv3\
     3963     ./sobin/gsc ./sobin/gsx -so-loader -so-loader -so-loader -so-loader
     3964    make[1]: Leaving directory '/mnt/local/cache/sbulut/spack-stage/spack-stage-ghostscript-10.04.0-i3zwzefgcjstngiszc77hyvncoamqlv3/spack-src'

See build log for details:
  /cache/sbulut/spack-stage/spack-stage-ghostscript-10.04.0-i3zwzefgcjstngiszc77hyvncoamqlv3/spack-build-out.txt
```
**may be ghostscript needs iconv dependency as of newer versions?**

--- a/repos/spack_repo/builtin/packages/mesa/package.py
+++ b/repos/spack_repo/builtin/packages/mesa/package.py
@@ -221,8 +221,13 @@ class MesonBuilder(meson.MesonBuilder):
             args.append("-Dgallium-omx=disabled")
 
         args_platforms = []
-        args_gallium_drivers = ["swrast"]
         args_dri_drivers = []
+
+        # swrast includes softpipe and llvmpipe
+        if spec.satisfied("+llvm"):
+            args_gallium_drivers = ["swrast"]
+        else:
+            args_gallium_drivers = ["softpipe"]
 
         opt_enable = lambda c, o: "-D%s=%sabled" % (o, "en" if c else "dis")
         opt_bool = lambda c, o: "-D%s=%s" % (o, str(c).lower())

--- a/repos/spack_repo/builtin/packages/mesa/package.py
+++ b/repos/spack_repo/builtin/packages/mesa/package.py
@@ -224,7 +224,7 @@ class MesonBuilder(meson.MesonBuilder):
         args_dri_drivers = []
 
         # swrast includes softpipe and llvmpipe
-        if spec.satisfied("+llvm"):
+        if spec.satisfies("+llvm"):
             args_gallium_drivers = ["swrast"]
         else:
             args_gallium_drivers = ["softpipe"]

